--- /proc/self/fd/11	2022-04-03 16:11:10.178931262 +0000
+++ kore_mustach.c	2022-04-03 16:10:52.188930266 +0000
@@ -340,9 +340,6 @@
         if ((item = kore_json_find(o, name, type)) != NULL)
             return (item);
 
-        if (kore_json_errno() != KORE_JSON_ERR_TYPE_MISMATCH)
-            return (NULL);
-
         type = type << 1;
     }
 
@@ -478,7 +475,7 @@
 
         case KORE_JSON_TYPE_INTEGER:
             i = kore_strtonum64(value, 1, &err);
-            return (!err) ? 0 : (o->data.integer > i) - (o->data.integer < i);
+            return (!err) ? 0 : (o->data.s64 > i) - (o->data.s64 < i);
 
         case KORE_JSON_TYPE_INTEGER_U64:
             u = kore_strtonum64(value, 0, &err);
@@ -579,9 +576,6 @@
 {
     size_t err = mustach_errno * -1;
 
-    if (mustach_errno == kore_json_errno())
-        return (kore_json_strerror());
-
     if (err < sizeof(mustach_errtab) / sizeof(mustach_errtab[0]))
         return (mustach_errtab[err]);
 
@@ -607,7 +601,6 @@
     mustach_errno = mustach_file(template, 0, &itf, &cl, flags, 0);
 
     if (mustach_errno >= 0) {
-        mustach_errno = kore_json_errno();
         *result = cl.result;
     } else {
         kore_buf_free(cl.result);
